client/my-sites/customer-home/free-photo-library-card/index.jsx

```diff
@@ -35,6 +39,16 @@ const FreePhotoLibraryCard = ( { recordTracksEvent: tracks } ) => {
 		return setShowDialog( ! showDialog );
 	};
 
+	const openSupportAndTrack = () => {
```

Comment:
Almost! At the moment this will functionally work, but the analytics call won't be made. This can be moved down to just above connect, so we can define this once vs during each render. We also need to add this function `openSupportAndTrack` to mapDispatchToProps.

eg
```jsx
const mapDispatchToProps = { openSupportAndTrack, recordTracksEvent }; // recordTracksEvent is still here since we directly call this in the other function.

export default  connect( null, mapDispatchToProps )( FreePhotoLibraryCard );
```

Reply:
Thanks @gwwar! I'm currently running into all sorts of errors when I try fixing the `Uncaught TypeError: openSupportArticleDialog is not a function` issue, including with

```jsx
const openSupportAndTrack = ( {
	recordTracksEvent: tracks,
	openSupportArticleDialog: supportArticleDialog,
} ) => {
	withAnalytics(
			tracks( 'calypso_customer_home_free_photo_library_video_support_page_view' ),
			supportArticleDialog( {
				postId: 145498,
				postUrl: localizeUrl( 'https://support.wordpress.com/free-photo-library/' ),
			} )
		);
	};
const mapDispatchToProps = { openSupportAndTrack, recordTracksEvent };

export default  connect( null, mapDispatchToProps )( FreePhotoLibraryCard );
```

Any idea on how to make that work? :)

Reply:
no need to pass through props in openSupportAndTrack. 
```jsx
const openSupportAndTrack = (  ) => { //no args, recordTracksEvent/openSupportArticleDialog is defined by the import at the top of file
	withAnalytics(
			tracks( 'calypso_customer_home_free_photo_library_video_support_page_view' ),
			supportArticleDialog( {
				postId: 145498,
				postUrl: localizeUrl( 'https://support.wordpress.com/free-photo-library/' ),
			} )
		);
	};
const mapDispatchToProps = { openSupportAndTrack, recordTracksEvent };

export default  connect( null, mapDispatchToProps )( FreePhotoLibraryCard );
```

There's a similar example in https://github.com/Automattic/wp-calypso/blob/8fbf16f0ab3be90a85a3d7dff56f093fc102be73/client/my-sites/hosting/sftp-card/index.js#L253 if that helps

Reply:
Thanks @gwwar! I'm still getting this error, which is becoming tricky to fix:

`index.jsx:91 Uncaught ReferenceError: tracks is not defined`

> feel free to leave it as two redux calls (what you originally had), and I can help with a janitorial after this lands. 👍

I think it's best to do this because this definitely is confusing me! I've updated the PR to how it was before. Thank you! :)